zerver/actions/users.py

```diff
@@ -280,108 +293,6 @@ def change_user_is_active(user_profile: UserProfile, value: bool) -> None:
-    notify_user_ids: list[int],
-) -> None:
-    realm = setting_group.realm
-    for setting_name in Realm.REALM_PERMISSION_GROUP_SETTINGS:
-        if getattr(realm, setting_name + "_id") == setting_group.id:
-            new_setting_value = UserGroupMembersData(
-                direct_members=group_members_dict[setting_group.id],
-                direct_subgroups=group_subgroups_dict[setting_group.id],
-            )
-            event = dict(
-                type="realm",
-                op="update_dict",
-                property="default",
-                data={setting_name: convert_to_user_group_members_dict(new_setting_value)},
-            )
-            send_event_on_commit(realm, event, notify_user_ids)
-            return
-
-
-def send_update_events_for_anonymous_group_settings(
-    setting_groups: list[UserGroup], realm: Realm, notify_user_ids: list[int]
-) -> None:
-    setting_group_ids = [group.id for group in setting_groups]
-    membership = (
-        UserGroupMembership.objects.filter(user_group_id__in=setting_group_ids)
-        .exclude(user_profile__is_active=False)
-        .values_list("user_group_id", "user_profile_id")
-    )
-
-    group_membership = GroupGroupMembership.objects.filter(
-        supergroup_id__in=setting_group_ids
-    ).values_list("subgroup_id", "supergroup_id")
-
-    group_members = defaultdict(list)
-    for user_group_id, user_profile_id in membership:
-        group_members[user_group_id].append(user_profile_id)
-
-    group_subgroups = defaultdict(list)
-    for subgroup_id, supergroup_id in group_membership:
-        group_subgroups[supergroup_id].append(subgroup_id)
-
-    group_setting_query = Q()
-    for setting_name in NamedUserGroup.GROUP_PERMISSION_SETTINGS:
-        group_setting_query |= Q(**{f"{setting_name}__in": setting_group_ids})
-
-    named_groups_using_setting_groups_dict = {}
-    named_groups_using_setting_groups = NamedUserGroup.objects.filter(realm=realm).filter(
-        group_setting_query
-    )
-    for group in named_groups_using_setting_groups:
-        for setting_name in NamedUserGroup.GROUP_PERMISSION_SETTINGS:
-            setting_value_id = getattr(group, setting_name + "_id")
-            if setting_value_id in setting_group_ids:
-                named_groups_using_setting_groups_dict[setting_value_id] = group
-
-    for setting_group in setting_groups:
-        if setting_group.id in named_groups_using_setting_groups_dict:
-            named_group = named_groups_using_setting_groups_dict[setting_group.id]
-            send_group_update_event_for_anonymous_group_setting(
-                setting_group,
-                group_members,
-                group_subgroups,
-                named_group,
-                notify_user_ids,
-            )
-        else:
-            send_realm_update_event_for_anonymous_group_setting(
-                setting_group,
-                group_members,
-                group_subgroups,
-                notify_user_ids,
```

Comment:
Added a new function for sending these events since we now have data for all the settings in `GroupPermissionUpdates`. Also, all realm settings are sent in a single event unlike what we did here with one event for each setting.